### PR TITLE
Duplo 16671 gcp application load balancer global access support

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

gcp application load balancer global access support
## Summary of changes

This PR does the following:

- gcp application load balancer global access support
- changed waiting api, created new function , purpose not to break other code using older waiting api

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
enhancement, documentation

___

## **Description**
- Introduced support for the `allow_global_access` attribute in GCP internal load balancer configurations, allowing users to specify global access settings.
- Added a new API endpoint `TenantGetLbDetailsInServiceNew` for enhanced load balancer details retrieval, improving the accuracy and reliability of state checks.
- Updated SDK and Terraform provider to support the new `allow_global_access` attribute, ensuring compatibility and extending functionality.
- Updated documentation to include the new `allow_global_access` attribute for both resources and data sources, providing clear guidance for users.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_lbconfigs.go</strong><dd><code>Support for Global Access in GCP Load Balancer Configurations</code></dd></summary>
<hr>
      
duplocloud/resource_duplo_lbconfigs.go

<li>Added <code>allow_global_access</code> attribute to load balancer schema.<br> <li> Implemented support for <code>allow_global_access</code> in load balancer <br>configurations.<br> <li> Updated state check logic in <code>duploServiceLbConfigsWaitUntilReady</code> <br>function.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-41a83bf1ecdb965c35671829b7fa60279941ef1a968e3410c2b9e2f285162e10">+24/-7</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>replication_controller.go</strong><dd><code>SDK Update for Global Access in Load Balancer Configurations</code></dd></summary>
<hr>
      
duplosdk/replication_controller.go

- Added `AllowGlobalAccess` field to `DuploLbConfiguration` struct.

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-06e27817a5fa8bc4ef2149b3caf90ba382fb422cdd7faf1ff225fe1c929a499e">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>tenant_aws_cloud_resources.go</strong><dd><code>New API Endpoint for Enhanced Load Balancer Details Retrieval</code></dd></summary>
<hr>
      
duplosdk/tenant_aws_cloud_resources.go

<li>Introduced <code>TenantGetLbDetailsInServiceNew</code> function for enhanced load <br>balancer details retrieval.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-92155358c5460b82348654a065ca4a047d7e3926595f74647c8786b1f7f02387">+13/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Changelog Update for Global Access Support and Enhancements</code></dd></summary>
<hr>
      
CHANGELOG.md

<li>Documented support for <code>allow_global_access</code> attribute and new API <br>endpoint.<br> <li> Updated entries for fixed issues and enhancements.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+26/-15</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>duplo_service_lbconfigs.md</strong><dd><code>Documentation for `allow_global_access` Data Source Attribute</code></dd></summary>
<hr>
      
docs/data-sources/duplo_service_lbconfigs.md

<li>Documented <code>allow_global_access</code> attribute in data source documentation.<br> <br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-3b2cc6c59d6e84889c7dd84f048dd01b1c0e02b48aa006e14f44393423855b4f">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>duplo_service_lbconfigs.md</strong><dd><code>Documentation for `allow_global_access` Resource Attribute</code></dd></summary>
<hr>
      
docs/resources/duplo_service_lbconfigs.md

<li>Added documentation for <code>allow_global_access</code> attribute in resource <br>documentation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/480/files#diff-1948136dba5174d211a7b5c40cd528de0c742563deae48e7df4bd31bd14ef307">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-16671 TF:support for global access attribute for ALB /Internal](https://app.clickup.com/t/8655600/DUPLO-16671)

File: duplocloud/resource_duplo_lbconfigs.go [hunk 1]
```diff
@@ -450,8 +450,14 @@ func duploServiceLbConfigsWaitUntilReady(ctx context.Context, c *duplosdk.Client
...
-				if lbDetails == nil {
-					return name, "pending", nil
+				if lbDetails != nil {
+					if val, ok := lbDetails["State"]; ok {
+						if val != "active" {
+							return name, "pending", nil
+
+						}
+					}
+					return name, "ready", nil
```
Comment (Duplo-nikv) [inline duplocloud/resource_duplo_lbconfigs.go:461]: response from lbdetails from gcp doesnt have State attribute, but aws has.
code logic explanation:
check if lbdetails has data
if source is aws then lbDetails will have State attribute if its not active poll
if source is gcp and lbdetails will not have State attribute, lbdetails only contains value once lb is active

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
+
+### Documentation
+- Corrected GKE credentials documentation and Terraform example, updating references from EKS to GKE and ensuring output values accurately reflect GKE credentials.
+
 markdown
 
 ## 2024-04-05
 
+### Added
+- Introduced support for `allow_global_access` attribute in GCP internal load balancer configurations.
```
Comment (komalkadam-dc) [inline CHANGELOG.md:21]: Please add the proper resource name

**Review by komalkadam-dc — CHANGES_REQUESTED** [review]
rest of the PR looks good, please fix changelog

File: CHANGELOG.md (hunk 2 above)
Comment (Duplo-nikv) [reply CHANGELOG.md]: done

**Review by komalkadam-dc — APPROVED** [review]

File: duplocloud/resource_duplo_lbconfigs.go [hunk 3]
```diff
@@ -149,6 +149,11 @@ func duploLbConfigSchema() map[string]*schema.Schema {
 			Type:        schema.TypeInt,
 			Computed:    true,
 		},
+		"allow_global_access": {
+			Description: "Applicable for internal lb.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+		},
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_lbconfigs.go:156]: Add computed here.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_lbconfigs.go:157]: done

File: duplocloud/resource_duplo_lbconfigs.go (hunk 1 above)
Comment (tahir-duplo) [reply duplocloud/resource_duplo_lbconfigs.go:462]: @Duplo-nikv I am not comfortable with this change, Have ou tested this for both AWS and GCP.

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**